### PR TITLE
Format `DrumSynth`

### PR DESCRIPTION
**Pull request by Monospace-V**
This file is so old I am making a separate pull request for it instead of grouping its cleaning with the rest of my categories.
Several issues here.
Bad formatting includes poor space usage (no space around operators / flow control), bad nextline usage and bad indentation (two space indentation instead of tab).

Additionally there is ugly logic (nested if loops).

I will fix as much of the _formatting_ as I can myself. I cannot speak for the rest.
Hopefully the next person approaching this file will have at least a better formatted task ahead of them.

### CODE REVIEW

**Comment by Monospace-V** [comment]
someone will want to run through this and check space usage where typecasting is done, and maybe glance through conventions?

File: src/core/DrumSynth.cpp [hunk 1]
```diff
@@ -63,22 +63,22 @@ short DD[1200], clippoint;
...
-	{
-		long eon = e - 1;
-		if (eon > 2) { eon = eon - 1; }
-		long p = 0;
-		while (envpts[e][0][p + 1] >= 0.f) { p++; }
-		envData[e][MAX] = envpts[e][0][p] * timestretch;
-		if (chkOn[eon] == 1 && envData[e][MAX] > l) { l = envData[e][MAX]; }
+  for (e = 1; e < 7; e++) //3
+  {
+    eon = e - 1; if (eon > 2) { eon = eon - 1; }
```
Comment (Rossmaxx) [inline src/core/DrumSynth.cpp:77]: The if could go in a different line

File: src/core/DrumSynth.cpp [hunk 2]
```diff
@@ -63,22 +63,22 @@ short DD[1200], clippoint;
...
+    p = 0;
+    while (envpts[e][0][p + 1] >= 0.f) { p++; }
+    envData[e][MAX] = envpts[e][0][p] * timestretch;
+    if (chkOn[eon] == 1) if (envData[e][MAX] > l) { l = envData[e][MAX]; }
```
Comment (Rossmaxx) [inline src/core/DrumSynth.cpp:81]:
```suggestion
    if (chkOn[eon] == 1 && envData[e][MAX] > l) { l = envData[e][MAX]; }
```

File: src/core/DrumSynth.cpp [hunk 3]
```diff
@@ -88,12 +88,12 @@ int DrumSynth::LongestEnv()
...
-  float loudest=0.f;
-  int i=0;
+  float loudest = 0.f;
+  int i = 0;
 
-  while (i<5) //2
+  while (i < 5) //2
   {
-    if(chkOn[i]==1) if(sliLev[i]>loudest) loudest=(float)sliLev[i];
+    if (chkOn[i] == 1) if (sliLev[i] > loudest) loudest = (float)sliLev[i];
```
Comment (Rossmaxx) [inline src/core/DrumSynth.cpp:96]:
```suggestion
    if (chkOn[i] == 1 && sliLev[i] > loudest) { loudest = (float)sliLev[i]; }
```

File: src/core/DrumSynth.cpp [hunk 4]
```diff
@@ -102,44 +102,45 @@ float DrumSynth::LoudestEnv()
...
-	envData[e][ENV] = envpts[e][1][static_cast<long>(envData[e][PNT] + 0.f)] * 0.01f; // this level
-	float endEnv = envpts[e][1][static_cast<long>(envData[e][PNT] + 1.f)] * 0.01f; // next level
-	float dT = envData[e][NEXTT] - static_cast<float>(t);
-	if (dT < 1.0) { dT = 1.0; }
-	envData[e][dENV] = (endEnv - envData[e][ENV]) / dT;
-	envData[e][PNT] = envData[e][PNT] + 1.0f;
+  float endEnv, dT;
+                                                             //0.2's added
+  envData[e][NEXTT] = envpts[e][0][(long)(envData[e][PNT] + 1.f)] * timestretch; //get next point
+  if(envData[e][NEXTT] < 0) envData[e][NEXTT] = 442000 * timestretch; //if end point, hold
```
Comment (Rossmaxx) [inline src/core/DrumSynth.cpp:108]: Add braces even for single line ifs, I ain't mentioning the other places.

File: src/core/DrumSynth.cpp [hunk 5]
```diff
@@ -148,41 +149,34 @@ void DrumSynth::GetEnv(int env, const char *sec, const char *key, QString ini)
...
-	bool inSection = false;
-	int len = 0;
+    stringstream is;
+    bool inSection = false;
+    char *line;
+    char *k, *b;
+    int len = 0;
 
-	char* line = static_cast<char*>(malloc(200));
+    line = (char*)malloc(200);
```
Comment (Rossmaxx) [inline src/core/DrumSynth.cpp:179]: Keep the static cast there.

File: src/core/DrumSynth.cpp [hunk 6]
```diff
@@ -191,46 +185,59 @@ int DrumSynth::GetPrivateProfileString(const char *sec, const char *key, const c
     QByteArray dat = f.readAll().constData();
     is.str(string(dat.constData(), dat.size()));
 
-    while (is.good()) {
-        if (!inSection) {
-            is.ignore( numeric_limits<streamsize>::max(), '[');
+    while (is.good())
```
Comment (Rossmaxx) [inline src/core/DrumSynth.cpp:188]: This block below this line could use better cleaning. Try inverting the condition and use `break`

**Review by Rossmaxx — COMMENTED** [review]
Some initial thoughts i got from the diff. Try solving these. I haven't included all of it.

**Comment by sakertooth** [comment]
You know, it would've taken zero effort to format the entire document if you wanted to satisfy our formatting conventions.

**Comment by Monospace-V** [comment]
> You know, it would've taken zero effort to format the entire document if you wanted to satisfy our formatting conventions.

Unfortunately, I am stupid.

**Comment by Monospace-V** [comment]
> You know, it would've taken zero effort to format the entire document if you wanted to satisfy our formatting conventions.

It is too late now. I have put in non-zero effort which cannot be backspaced. I shall continue to do so. But would be kinda useful if you could tell me more about this

**Comment by sakertooth** [comment]
> It is too late now. I have put in non-zero effort which cannot be backspaced. I shall continue to do so. But would be kinda useful if you could tell me more about this

There are probably other ways, but using VSCode with our `.clang-tidy` file allows you to format the entire document based on what is specified there. I can push a commit to format everything here for you if you want.

**Comment by sakertooth** [comment]
>  At this point, if it does that, why don't we do everything with that instead?

Usually, we don't do this because there are a lot of out of scope changes it create when making PRs. But since this is a formatting PR, all the changes in here would be relevant and in scope.

**Comment by Monospace-V** [comment]
> Usually, we don't do this because there are a lot of out of scope changes it create when making PRs. But since this is a formatting PR, all the changes in here would be relevant and in scope.

Wait. All those PRs I plan to do to clean and there's a command I could do that just wasn't being done?
Can you (on discord maybe) just help me understand how it's done. Is it an option to go through and format a lot of files with this so everything's up to format?

**Comment by Monospace-V** [comment]
I notice that some of the block brace work has been deleted, among other things. This would mostly format whitespaces and ordering.
image
image
@sakertooth should I redo some of that work on this directly, or would you run that command on my last commit with that stuff done?
In the future I'll probably want to run that first and then go on with looking through.

**Comment by sakertooth** [comment]
Let me know if there's anything else that should be done here :+1:.

**Comment by Monospace-V** [comment]
It looks about right at first glance. I will check it from PC at a later point. Thank you.
Can you at some point explain to me what you did and direct me to the applicable documentation for the same?

**Comment by sakertooth** [comment]
>  Can you at some point explain to me what you did and direct me to the applicable documentation for the same?

Our `.clang-tidy` and `.clang-format` files pretty much make up what was done here (though the transition to use the named casts instead of C-style casts was done manually). You can take a look there for some ideas. Here are some details on [clang-tidy](https://clang.llvm.org/extra/clang-tidy/) and [clang-format](https://clang.llvm.org/docs/ClangFormat.html).

**Comment by Monospace-V** [comment]
I notice you have moved the asterisks. I was under the impression that `char *key` and `char* key` meant different things, but I believe you know what you are doing. I just want to bring this to your attention to confirm.

Additionally I perceive some confusing nextline usage/requirement I will probably want to resolve. I will not be at my home computer for about two or three days, so unlikely to do it till then. I will try to see if I can do something about it anyway.

I am under the understanding this pull request still be marked as ready for review due to your tendency to keep it that way.

**Comment by Veratil** [comment]
> I was under the impression that `char *key` and `char* key` meant different things

They actually are the same in this context. This confusion is why we put the pointer with the type instead of with the variable name.

**Comment by sakertooth** [comment]
> Additionally I perceive some confusing nextline usage/requirement I will probably want to resolve. I will not be at my home computer for about two or three days, so unlikely to do it till then. I will try to see if I can do something about it anyway.

Feel free to do anything remaining that you think fits with our conventions, I just wanted to help you out a bit since it seemed like you were facing some difficulty with this file. :+1:

**Comment by Rossmaxx** [comment]
The file seems to be quite clean now. If no one objects or finds any flaws, i'll merge in 2 days.

**Review by Veratil — APPROVED** [review]

**Pull request merged**